### ISSUE TITLE
Blank Page when Trying to Submit

### ISSUE THREAD

**Issue by Dara-K** [topcoder-platform/community-app#4144]
**Describe the bug**
When members are trying to submit on Design Challenges they are seeing a blank page as per attached document.

**To Reproduce**
The bug was reported for this challenge (checkpoint phase) https://www.topcoder.com/challenges/30120209 but I was told it happens to other challenges as well for other members.
1. Press on Submit button.
2. The Submit page will show a blank page then

**Expected behavior**
The Submit page should show the option for a member to upload his file.

**Screenshots**
![thumbnail](https://user-images.githubusercontent.com/25185079/77848814-1d7ba480-71d0-11ea-8946-e711c6a0b7de.png)

**Desktop (please complete the following information):**
 - OS: Mac OS Catalina, version 10.15.2
 - Browser: Google Chrome, Version  80.0.3987.149 (Official Build) (64-bit)

**Additional context**
The member said "main issue is that the submit page refreshes by itself even if I drag and drop files for upload or use any cloud option. The page itself disappears to a blank screen."

Hope we can solve this soon as it's affecting several members and it creates a bad experience on the site.

The work should be down against `hot-fix-submit-blank` branch

**Comment by Dara-K** [comment]
Adding @jonimazumder to the conversation as well

**Comment by SathyaJayabal** [comment]
@Dara-K , is this. still happening? I was able to submit to a design challenge(not the one mentioned above though)

**Comment by Dara-K** [comment]
Yes, is happening - this happened yesterday to one member

**Comment by SathyaJayabal** [comment]
@Oanh-and-only-Oanh , This issue seems to be happening to some users. I was not able to reproduce it.
cc @Dara-K

**Comment by Dara-K** [comment]
What can we do to help here? @Oanh-and-only-Oanh @SathyaJayabal

**Comment by Oanh-and-only-Oanh** [comment]
@Dara-K, we need to be able to reproduce this bug in order to fix it. The best is if a member faces the issues, they should record their screen

**Comment by Dara-K** [comment]
Noted on that @Oanh-and-only-Oanh . Should the member keep the browser console open when doing so or anything similar?

**Comment by LieutenantRoger** [comment]
> Noted on that @Oanh-and-only-Oanh . Should the member keep the browser console open when doing so or anything similar?
Hi, @Dara-K  yes, it's better to include the console in the recording as well if the console's showing errors.


I seemed not able to reproduce, please see the recording: https://monosnap.com/file/WmSm0ZZuyOqTsxcV0ogAWcE0nv0EYx

**Comment by LieutenantRoger** [comment]
Though I found another issue related to console on a design challenge detail page here: 
https://github.com/topcoder-platform/community-app/issues/4171

@Dara-K it looks different with your one created here.

**Comment by Dara-K** [comment]
Thanks @LieutenantRoger  for looking into this. I let the  @jonimazumder  know to share a video recording for us next time when submitting.

I see the other error report. Not sure how it's affecting the challenge though. Hope it can be fixed.

**Comment by jonimazumder** [comment]
Hello Friends,

I have attached drive link below where you can see the exact problem that I am facing.
There are two videos.
1. Submit page blank screen.mov- As discussed earlier the blank screen appears again and again and self refreshes every time after2-3 seconds on its own.
2. Challenge page refreshes by itself, not able to view my challenges. It shows up for 2-3 seconds and self refreshes.

Please find the video by clicking below video:
https://drive.google.com/drive/folders/13pvf1MpKBYBqFJro1NQX3XWBryBERoAj?usp=sharing

**Comment by jonimazumder** [comment]
Hello,

Please find the blank submit screen, Photos with console code shown.
https://drive.google.com/drive/folders/1GBo2JtQMhBrW4zHhjN7ysrUn4rO2UPKn?usp=sharing

Is video of the same screen is required along with console ?

**Comment by LieutenantRoger** [comment]
Hi @jonimazumder 

Thanks for the detail videos and the console logs.  These are detail enough for us. :) 

For the 1
I can see you are filtering the challenges on challenge list screen for design track, filtering challenges are not working well with V3 challenge APIs, we have been in the process of using the new V5 challenge API now. And if the issues remained with V5 API, we will try to resolve it against V5 API. 

So for now, I would suggest to not include filters in the urls when visiting the challenge list screen. 
Also I think you can use the link on your Online Review system, to navigate to a challenge detail page you have registered. For example:  https://take.ms/QYFuR

For the 2

I looks you are visiting the `\submit` page without login before.  This is a valid issue. 
For now, as a work around, please make sure you have login before visiting `\submit` page for your challenge.  

I think above work-arounds would meet your needs for now.
Meanwhile, we will look into these issues.

**Comment by jonimazumder** [comment]
Hi @LieutenantRoger For blank screen issue. I am logging in that scenario but on pressing submit button it logouts automatically

**Comment by jonimazumder** [comment]
Please do some work around for the blank screen issue as I am not able to submit to any challenge and its crucial.

**Comment by LieutenantRoger** [comment]
I'm not sure your scenario, see the submission page on my side: https://monosnap.com/file/kmKMoEO5lBdCeMae0DPMphV0SvgYxq

Maybe you could 
1) Try another browser: e.g. firefox to perform the submit operation.  
2) Try to use your Chrome in incognite mode, I suspect it might be relevant with your current Chrome setting. 

3) As a always working work-around,  please email to support@topcoder.com with your submission and corresponding challenge, the support team will always be able to help you to make the submission.

**Comment by LieutenantRoger** [comment]
Hello @Oanh-and-only-Oanh 

It's strange with the @jonimazumder 's environment that clicking `submit` button on challenge detail screen would logout him.

But as a general solution,

We can show this screen if a member is not login but managed to reach the `/submit` page of a challenge

![image](https://user-images.githubusercontent.com/2875203/78473146-d9a91200-7770-11ea-88db-33ece7cea038.png)


Please let me know if this is fine.

**Comment by jonimazumder** [comment]
@LieutenantRoger 
Even I tried in Incognito of Chrome and firefox browser. But problem is still there.
After clicking the submit button from the challenge screen. Blank screen comes.

**Comment by LieutenantRoger** [comment]
@jonimazumder  Hmmm, it's strange, I could only reproduce the blank screen without login to the site.  

Have you run into this screen before ?   When I login, I can submit without problem:  see the recording I have shared above: https://monosnap.com/file/kmKMoEO5lBdCeMae0DPMphV0SvgYxq

So I can't be sure if this is an issue on our site except the confirmed issues above. 

I suggest you to take option 3, which is to ask help from support by emailing support@topcoder.com with your submission as the attachment and with the corresponding challenge link, brief the reason that you can't make a submission and ask them to help you.

**Comment by jonimazumder** [comment]
[Blank screen.docx](https://github.com/topcoder-platform/community-app/files/4433581/Blank.screen.docx)

**Comment by jonimazumder** [comment]
@LieutenantRoger 
I have already mailed to support.topcoder way back, but issue was not resolved.

I guess the main issue is that the site refreshes by itself and then presented with total blank screen

**Comment by jonimazumder** [comment]
502 Bad Gateway

this message appears on the blank screen

**Comment by jonimazumder** [comment]
@LieutenantRoger 
I have seen the video of your account "Submit screen" which stays normal and doesnot refreshes by itself.
For me the refreshes thing happens and blank screen comes up.

**Comment by LieutenantRoger** [comment]
Glad that you have contact support,  thanks for the `Blank screen.docx`.

I have resolved related console errors, the result is mentioned here https://github.com/topcoder-platform/community-app/issues/4144#issuecomment-609400617

Once we confirmed that, I can start the process of pushing the fix to production.

I think refreshing issue should be another issue, we will try to fix that after we completed the migration onto V5 challenge API.

**Comment by jonimazumder** [comment]
Thankyou @LieutenantRoger 
So is the problem identified, like what went wrong.
And why its only appearing for me.

**Comment by LieutenantRoger** [comment]
Like mentioned above, I haven't been able to reproduce this sceario:  "After login, then click submit button to reach `/submit` page would lead to a blank screen"

But I have reproduced this scenario: "without login, and reach the `/submit` page would lead to a blank screen", so right now I would expect my fix might fix your scenario as well which I couldn't reproduce.

In general, now you shouldn't be blocked with making submission, right ?  :)

**Comment by jonimazumder** [comment]
Appreciate your efforts @LieutenantRoger.

ok got it.

But any idea why its only happening to me and not you or many others.
like is there any site issue?

**Comment by LieutenantRoger** [comment]
No idea from my side now, we will let you know after we deploy this change. https://github.com/topcoder-platform/community-app/issues/4144#issuecomment-609400617 and see if you are still encountering the issue.

**Comment by jonimazumder** [comment]
ok @LieutenantRoger 

Hoping and praying that things get fixed soon. Actually these are causing a great deal of troubles for me.
Wasting a huge amount of time on mailing back and forth for submission related issues and in even viewing the challenges.

Its like after opening the challenge screem i have only got 2-3 seconds and i am shooting on challenge links. or they are lost for another shutdown ...hahahaa :)

Thanks a lot for your efforts.

**Comment by LieutenantRoger** [comment]
ok, you are welcome.

Like mentioned above, always use this link to view the challenge list screen for now, 
`https://www.topcoder.com/challenges`  (without any filters applied)

**Comment by Dara-K** [comment]
Thank you for looking into this @LieutenantRoger . Could you let us know here when the fix for #4144 will be deployed?

**Comment by LieutenantRoger** [comment]
@Dara-K  once we confirmed the result mentioned here https://github.com/topcoder-platform/community-app/issues/4144#issuecomment-609400617 ,  we can start to deploy it, the deploying process will cost 1 or 2 days.

**Comment by Oanh-and-only-Oanh** [comment]
> Hello @Oanh-and-only-Oanh
> 
> It's strange with the @jonimazumder 's environment that clicking `submit` button on challenge detail screen would logout him.
> 
> But as a general solution,
> 
> We can show this screen if a member is not login but managed to reach the `/submit` page of a challenge
> 
> ![image](https://user-images.githubusercontent.com/2875203/78473146-d9a91200-7770-11ea-88db-33ece7cea038.png)
> 
> Please let me know if this is fine.

Let's change the message to say "You must be logged in to access this page"

**Comment by LieutenantRoger** [comment]
Hi @Oanh-and-only-Oanh 

Actually I think current text is better.

The real criteria for accessing the submit page is that a member is login AND reigstered the corresponding challenge.

So login only won't be The only criteria For accessing the submit page.
What do you think ?

**Comment by Oanh-and-only-Oanh** [comment]
Hm, we need to give more direction to the user. How about "You are not authorized to access this page" and give option for user to "Go to Challenge Details"?

**Comment by LieutenantRoger** [comment]
Ok, added a button to allow member navigate back to challenge detail page like this: 

![image](https://user-images.githubusercontent.com/2875203/78546362-d88aea00-782f-11ea-979d-2473423742b3.png)

**Comment by jonimazumder** [comment]
I checked today but blank screen is still coming. And in console it says Topcoder API blocked by CORS Policy (what does that mean?). I am attaching in a doc about this api loading issue.

[Check error.docx](https://github.com/topcoder-platform/community-app/files/4437691/Check.error.docx)

**Comment by jonimazumder** [comment]
<img width="1459" alt="Screenshot 2020-04-06 at 3 50 02 PM" src="https://user-images.githubusercontent.com/20997307/78548676-cc4a6100-781e-11ea-83d4-0ff125bf1632.png">

**Comment by LieutenantRoger** [comment]
We just started to test the blank screen fix, we will push it to production ASAP, please be more patient.

Hmmm, I'm not able to reproduce the cross origin error, as well, please see my console with my own account, it's strange. 
![image](https://user-images.githubusercontent.com/2875203/78548946-2b66a080-7834-11ea-9d44-c65d3bc59f18.png)

**Comment by LieutenantRoger** [comment]
@jonimazumder  are you in a network that has some restriction ? If so, you will have to setup a VPN.

**Comment by sandhiyakavi** [comment]
Tested in dev environment.Working as expected.

<img width="960" alt="error_submiy" src="https://user-images.githubusercontent.com/55895538/78551191-ed14b580-7822-11ea-9da7-c87c6e93a900.PNG">

**Comment by jonimazumder** [comment]
@LieutenantRoger  I am checking on the VPN and behaviour after connecting to other networks

**Comment by SathyaJayabal** [comment]
verified on beta
<img width="1423" alt="Screenshot 2020-04-10 at 9 37 34 AM" src="https://user-images.githubusercontent.com/58783823/78961414-fb0e5300-7b0e-11ea-9b6e-1853058c34a7.png">

**Comment by Jobee1** [comment]
**Verified on Beta Environment:**
![4144 beta](https://user-images.githubusercontent.com/48575864/79132701-836c4c80-7dca-11ea-9234-ed96dc509783.png)

**Verified on Production Environment:**
![Screenshot_2](https://user-images.githubusercontent.com/48575864/79132728-8e26e180-7dca-11ea-9148-17d82e22c74d.png)

**Comment by SathyaJayabal** [comment]
@jonimazumder, are you still facing the issue of blank screen when trying to submit.

**Comment by SathyaJayabal** [comment]
@LieutenantRoger @Dara-K @Oanh-and-only-Oanh , correct message is displayed when user accesses the submit page without logging in/ registering.
Can we close the issue. The other issues reported by @jonimazumder are not reproducible by QA or. @LieutenantRoger. Please advice.

**Comment by jonimazumder** [comment]
@SathyaJayabal  the issue is still there. Just checked but again getting blank screen problem

**Comment by jonimazumder** [comment]
@SathyaJayabal @LieutenantRoger  I even tried with tor but issue still there.

**Comment by LieutenantRoger** [comment]
![image](https://user-images.githubusercontent.com/2875203/79243450-36f73e80-7ea8-11ea-8c61-5940d341c057.png)

![image](https://user-images.githubusercontent.com/2875203/79243769-a79e5b00-7ea8-11ea-8c3c-8840a883415c.png)

Hi @jonimazumder  I can't reproduce the blank screen problem.  But I believe the console error has been resolved after the fix.  

May I know which Country you are in ?  I'm based in China. I normally will connect VPN when visit TC site to make sure everything is loaded smoothly and correctly. :)

**Comment by Dara-K** [comment]
Can you try accessing the Topcoder in incognito mode too @jonimazumder ?

@LieutenantRoger I know we are solving site problems here, but i was wondering if we can help Joni by maybe working closer to him, 1-1? maybe connecting to his computer and seeing what the issue is would help better? at least to find a work around for him to submit as he's emailing support nowadays for every challenge he wants to submit, for both rounds.

As designers, we don't have too much development insight and knowledge, but you know so much more than us, so that would be greatly appreciated. :)

Is there any other thing we can do for this? other ideas?

**Comment by LieutenantRoger** [comment]
@Dara-K  do other members from design challenges receiving issues like @jonimazumder  ? 

If No, then there must be some local setup either computer or network on @jonimazumder 's environment.

**Comment by harshitmehta123** [comment]
Hey @LieutenantRoger I will connect 1:1 with @jonimazumder and find what the issue is to help you better. Will get back here on the issue in 2/3 days :)

**Comment by Dara-K** [comment]
Thank you @harshitmehta123!
 
@jonimazumder Can you write here which are the other members you said are facing same issue please?

**Comment by lakshmiathreya** [comment]
@SathyaJayabal - is this Issues resolved - in that case pls close it. Else move it to a later milestone? Thx!

**Comment by Oanh-and-only-Oanh** [comment]
placing on hold because "A certain user is not able to submit, but we are not able to reproduce the problem. Harshit has offered to work with the user one on one and try to find out more. No one else has the issue. But that user has been mailing support for everytime they submit."

**Comment by SathyaJayabal** [comment]
https://github.com/topcoder-platform/community-app/issues/4144#issuecomment-614046150

**Comment by jonimazumder** [comment]
@Dara-K  in one of the challenge forums egislr also mentioned the same issue

**Comment by Dara-K** [comment]
Thanks @jonimazumder !

@SathyaJayabal  I am attaching the screenshot Joni shared with me by email
![image](https://user-images.githubusercontent.com/25185079/79951297-19663e00-8481-11ea-837d-0935347694d2.png)
